exporter/exporterhelper/queued_retry.go

```diff
@@ -0,0 +1,196 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package exporterhelper
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/cenkalti/backoff"
+	"github.com/jaegertracing/jaeger/pkg/queue"
+
+	"go.opentelemetry.io/collector/consumer/consumererror"
+)
+
+// QueuedSettings defines configuration for queueing batches before sending to the nextSender.
+type QueuedSettings struct {
+	// Disabled indicates whether to not enqueue batches before sending to the nextSender.
+	Disabled bool `mapstructure:"disabled"`
+	// NumWorkers is the number of consumers from the queue.
+	NumWorkers int `mapstructure:"num_workers"`
+	// QueueSize is the maximum number of batches allowed in queue at a given time.
+	QueueSize int `mapstructure:"queue_size"`
+}
+
+func CreateDefaultQueuedSettings() QueuedSettings {
+	return QueuedSettings{
+		Disabled:   false,
+		NumWorkers: 10,
+		QueueSize:  5000,
+	}
+}
+
+// RetrySettings defines configuration for retrying batches in case of export failure.
+// The current supported strategy is exponential backoff.
+type RetrySettings struct {
+	// Disabled indicates whether to not retry sending batches in case of export failure.
+	Disabled bool `mapstructure:"disabled"`
+	// InitialBackoff the time to wait after the first failure before retrying
+	InitialBackoff time.Duration `mapstructure:"initial_backoff"`
+	// MaxBackoff is the upper bound on backoff.
```

Comment:
```suggestion
	// MaxBackoff is the upper bound on backoff. Once this value is reached, the data is discarded.
```

Reply:
Probably the terms confused you, that is the MaxElapsedTime that drops the data.

Reply:
The MaxBackoff is the maximum time that we will wait between retries. This is an exponential backoff strategy so we start from `InitialBackoff` and exponentially increase until `MaxBackoff` (using some jitter to ensure some randomness). Once we hit `MaxBackoff` we will always wait that much until we succeed or `MaxElapsedTime` expired.

Reply:
So @tigrannajaryan do you want to go with "InitialInterval" and "MaxInterval"?

Reply:
>do you want to go with "InitialInterval" and "MaxInterval"

Yes, I think this would be most understandable for users.